lib/pbench/test/unit/server/test_requests.py

```diff
@@ -130,7 +131,9 @@ def delete(self):
                 TestUpload.tarball_deleted = self.name
 
             def get_metadata(self):
-                return {"pbench": {"date": dateutil.parser.parse("2002-05-16")}}
+                return {
+                    "pbench": {"date": IsoTimeHelper.from_string("2002-05-16").utc_time}
```

Comment:
Do you really want to use `IsoTimeHelper` here, as opposed to `datetime.datetime.isoformat()`, like you did in `lib/pbench/test/unit/server/test_datasets_list.py`?  (Should the two be using the same approach?)

Reply:
The compelling (er, to the extent that it is...) reason would be to minimize the amount of project code used to test each unit (to avoid unanticipated interactions between disparate components).  That is, there is no compelling reason to use `IsoTimeHelper` when the standard library function will suffice.  (But, if you're not inclined to change this, then I'm not feeling compelled to argue.)

Reply:
> reason would be to minimize the amount of project code used to test each unit

Except that this is a mock masquerading as product code, so the cut is a bit more subtle than that. I liked the idea of applying the same mechanism.